cpp/src/io/comp/io_uncomp.h

```diff
@@ -38,9 +42,8 @@ enum {
 
 std::vector<char> io_uncompress_single_h2d(const void* src, size_t src_size, int stream_type);
 
-std::vector<char> getUncompressedHostData(const char* h_data,
-                                          size_t num_bytes,
-                                          const std::string& compression);
+std::vector<char> get_uncompressed_data(host_span<char const> const& data,
```

Comment:
view types are typically pass by value (`column_view` is an annoying exception). 
```suggestion
std::vector<char> get_uncompressed_data(host_span<char const> data,
```

Reply:
I recall seeing the guidelines about `column_view const&`, but why is it the exception?

Reply:
The short version is because `column_view` has a `std::vector` member. Not only does that make it not trivially copyable, but also breaks some things wrt to the `type_dispatcher`, `host/device` decoration, and nvcc compiler warnings that get promoted to errors.